components/tidb_query_executors/src/runner.rs

```diff
@@ -378,6 +378,8 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
     ) -> Result<Self> {
         let executors_len = req.get_executors().len();
         let collect_exec_summary = req.get_collect_execution_summaries();
+        let mut config = EvalConfig::from_request(&req)?;
+        config.paging_size = paging_size;
         let config = Arc::new(EvalConfig::from_request(&req)?);
```

Comment:
You should use the `config` created above instead of creating an `EvalConfig` again